packages/ui/src/views/execution/execution-status.tsx

```diff
@@ -4,15 +4,15 @@ import { ExecutionState } from '@views/repo/pull-request'
 import { BadgeProps, ExecutionStatusProps } from './types'
 
 const Badge: React.FC<ExecutionStatusProps & BadgeProps> = props => {
-  const { status, duration, minimal } = props
+  const { status, duration, minimal, inConnector = false } = props
   switch (status.toLowerCase()) {
     case ExecutionState.WAITING_ON_DEPENDENCIES:
     case ExecutionState.BLOCKED:
     case ExecutionState.PENDING:
       return minimal ? (
         <div className="flex items-center gap-1">
           <div className="size-2 rounded-full bg-cn-background-softgray" />
-          <span className="text-cn-foreground-disabled">Pending</span>
+          <span className={inConnector ? 'text-cn-foreground-1' : 'text-cn-foreground-disabled'}>Pending</span>
```

Comment:
I feel having an entity specific logic (entity being a connector) in a common component like this would scale, without having complex `if-else`s.

Reply:
so i shud make a seperate status component for this ? @vardanbansal-harness

Reply:
Discussing offline